### DISCUSSION TITLE
Meson build

### DISCUSSION THREAD

**Discussion by jwellbelove** [ETLCPP/etl#435] [Q&A]
@phillipjohnston 
I was playing around with using meson for the  first time, but I am having an issue with the UnitTest++ dependency.

Running meson I get...
`Build-time dependency unittest++ found: NO (tried pkgconfig and cmake)`

Any ideas on how to get around this?

**Comment by phillipjohnston** [comment]
Hey @jwellbelove, what OS are you using, and how have you installed UnitTest++?

On MacOS with unittest-cpp installed from homebrew, there is a pkgconfig file so it "just works".

**Comment by jwellbelove** [comment]
I'm using Ubuntu under WSL.
The WSL instance accesses my Windows ETL code directories directly, so UnitTest++ is already there.
I'm not really much of a Linux guy, so I'm not sure how UnitTest++ would be 'installed' on it.

**Comment by jwellbelove** [comment]
OK, fixed it.
`sudo apt install libunittest++-dev`

**Comment by jwellbelove** [comment]
@phillipjohnston 
I have a pull request to change the meson build files.
As you use meson, could I ask you to give it a quick look to see if the changes makes sense to you?

hotfix/skorokhod-meson_tests_fix

**Comment by phillipjohnston** [comment]
The changes look good to me and I think are a good build improvement overall.

Cheers,

Phillip
